### PR TITLE
Decouples rigid object and articulation asset classes

### PR DESCRIPTION
**Pull request by Mayankm96**
# Description

Since we override a lot of the functions from RigidObject inside the Articulation class, we don't need to rely on inheritance anymore. Duplicacy in the code makes it easier to understand the two classes' functionalities without severely added overhead from the maintenance side. Moreover, conceptually, it can be motivated that the two are independent concepts.

This MR decouples the rigid object and articulation concepts in the framework.

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

File: source/extensions/omni.isaac.lab/test/assets/test_articulation.py
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab/test/assets/test_articulation.py]:
```suggestion
```

File: source/extensions/omni.isaac.lab/omni/isaac/lab/assets/articulation/articulation.py
Comment (jtigue-bdai) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/assets/articulation/articulation.py]: Does this need a doc string?

My interpretation of "num_instances" in this case is the number of replications of the articulation. Is this the correct interpretation? Can this be different than the number of environments?

**Review by jtigue-bdai — APPROVED** [review]
Everything looks good. I just have a specific question about the num_instances property of the articulation

File: source/extensions/omni.isaac.lab/omni/isaac/lab/assets/articulation/articulation.py
Comment (Mayankm96) [reply source/extensions/omni.isaac.lab/omni/isaac/lab/assets/articulation/articulation.py]: For most learning setup, "num_instances" = "num_envs". At least that's the main underlying idea. However, for a single environment case, the view can be used to capture two instances of the same robot in that environment.

Scaling it to N environments with M same robots in each remains a discussion point because all the tensor shapes change in that case. We can think of adding it if we see a huge perf gain by doing so, i.e. one view capturing M robots vs. one view per robot.

Comment (jtigue-bdai) [reply source/extensions/omni.isaac.lab/omni/isaac/lab/assets/articulation/articulation.py]: That sounds reasonable. What about learning setups for multi-robot collaboration (i.e. two arm manipulation) do we set it up such that the two robots are viewed as one?

File: source/extensions/omni.isaac.lab/omni/isaac/lab/assets/articulation/articulation.py
Comment (jsmith-bdai) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/assets/articulation/articulation.py:35]: Can you do a Ctrl-F for any instances of rigid + object and remove / update them? The docstring has some references to it being a subclass

**Comment by jsmith-bdai** [comment]
This seems like a big change, do you mind outlining the pros / cons of this approach?

It looks like we're not getting much benefit at first glance, but are now having a lot of repeated code that will be difficult to maintain

**Comment by Mayankm96** [comment]
Will work on this MR once #760 and #742 are merged.

File: source/extensions/omni.isaac.lab/omni/isaac/lab/assets/articulation/articulation.py
Comment (Mayankm96) [reply source/extensions/omni.isaac.lab/omni/isaac/lab/assets/articulation/articulation.py]: There are two separate robot objects. We don't support them as "one" articulation view.

**Review by Dhoeller19 — APPROVED** [review]

File: source/extensions/omni.isaac.lab/omni/isaac/lab/assets/articulation/articulation.py
Comment (Mayankm96) [reply source/extensions/omni.isaac.lab/omni/isaac/lab/assets/articulation/articulation.py:35]: Done.

**Review by jsmith-bdai — APPROVED** [review]
LGTM if sufficiently tested!

We need to keep an eye on any of the now duplicated code changing and make sure it is changed in both places (although these changes seem unlikely which is good!)

**Pull request merged**
